dsl/src/commonMain/kotlin/com/freeletics/flowredux/dsl/internal/CollectInStateSideEffectBuilder.kt

```diff
@@ -38,15 +39,15 @@ internal class CollectInStateBuilder<T, InputState : S, S : Any, A : Any>(
     override fun generateSideEffect(): SideEffect<S, Action<S, A>> {
         return { actions: Flow<Action<S, A>>, getState: GetState<S> ->
             actions
-                .filterState(getState = getState, isInState = isInState)
-                .flatMapLatest { stateSubscription ->
-                    when (stateSubscription) {
-                        FilterState.StateChanged.SUBSCRIBE -> {
-                            flow.flatMapWithPolicy(flatMapPolicy) {
-                                setStateFlow(value = it, getState = getState)
-                            }
+                .map { isInState(getState()) }
+                .distinctUntilChanged()
```

Comment:
Similarly to last Time with ˋflowOfCurrentState()ˋ: l would extract this into another function to increase readability...

Reply:
Extracted it into an operator called `mapToIsInState`. Also using that in `onEnter`